File: 프로그래머스/lv2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.java
import java.util.HashSet;
import java.util.Arrays;
import java.util.Set;
class Solution {
    public int solution(int[] elements) {
        int[] newElements=new int[elements.length*2];
    for(int i=0;i<newElements.length;i++){
      newElements[i]=i<elements.length?elements[i]:elements[i-elements.length];
    }


    Set<Integer> list=new HashSet<>();
    for(int i=1;i<=elements.length;i++){
      for(int j=0;j<elements.length;j++){
          list.add(Arrays.stream(newElements,j,j+i).sum());
      }
    }


    return list.size();
    }
}

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    answer = 0
    li = set()
    for i in range(len(elements)):
        v = elements[i]
        li.add(v)
        for j in range(i+1, i+len(elements)):
            v+=elements[j%len(elements)]
            li.add(v)
    return len(set(li))